### PR TITLE
Fix wrong references of document

### PR DESCRIPTION
**Pull request by not522**
I fix wrong references using nitpicky mode of Sphinx.
Enable nitpicky mode is suggested (#657), but it is a bit difficult to fix all warnings.
Therefore, I first write PR about fixing easy errors.

### CODE REVIEW

File: cupy/cuda/__init__.py [hunk 1]
```diff
@@ -62,6 +62,7 @@ def is_available():
 from cupy.cuda.memory import MemoryPointer  # NOQA
 from cupy.cuda.memory import MemoryPool  # NOQA
 from cupy.cuda.memory import set_allocator  # NOQA
+from cupy.cuda.memory import SingleDeviceMemoryPool  # NOQA
```
Comment (okuta) [inline cupy/cuda/__init__.py]: I want to hide this class.

File: docs/source/reference/cuda.rst [hunk 2]
```diff
@@ -21,12 +21,16 @@ Memory management
...
+   cupy.cuda.PinnedMemory
    cupy.cuda.MemoryPointer
+   cupy.cuda.PinnedMemoryPointer
    cupy.cuda.alloc
+   cupy.cuda.alloc_pinned_memory
    cupy.cuda.set_allocator
    cupy.cuda.set_pinned_memory_allocator
    cupy.cuda.MemoryPool
    cupy.cuda.PinnedMemoryPool
+   cupy.cuda.SingleDeviceMemoryPool
```
Comment (okuta) [inline docs/source/reference/cuda.rst]: ditto.

**Comment by okuta** [comment]
jenkins, test this please.

File: docs/source/reference/cuda.rst (hunk 2 above)
Comment (not522) [reply docs/source/reference/cuda.rst]: Do you want to hide it on the document?
It is referred from `cupy.cuda.MemoryHook` now.

Which do you think better?

- Remove reference to `SingleDeviceMemoryPool`.
- Refer to `cupy.cuda.memory.SingleDeviceMemoryPool`.

In second option we can hide it on the list, it is just referred from `MemoryHook`.

Comment (okuta) [reply docs/source/reference/cuda.rst]: OK, Cloud you fix `MemoryHook` document?
I think that the link to `MemoryPool` is enough for that explanation.

Comment (not522) [reply docs/source/reference/cuda.rst]: OK, I skip it here, and make another PR to fix `MemoryHook` document.
I want to treat only easy reference errors in this PR.

**Comment by okuta** [comment]
I respect your decision.
LGTM.

**Pull request merged**